### PR TITLE
MEPTS-926 | TX_RTT change request Mer 2.6

### PR DESCRIPTION
**Pull request by iteliolucas**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java [hunk 1]
```diff
@@ -604,4 +532,76 @@ public CohortDefinition getTreatmentInterruptionOfXDaysBeforeReturningToTreatmen
...
+            "onOrAfter=${startDate},onOrBefore=${endDate},location=${location}"));
+
+    cd.addSearch(
+        "AAB",
+        EptsReportUtils.map(
+            getTreatmentInterruptionOfXDaysBeforeReturningToTreatment(minDays, maxDays, 28),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString(
+        "initiatedPreviousPeriod AND returned AND txcurr AND (LTFU AND NOT (transferredOut OR transferredIn)) AND AAB");
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java:603]: The specs calls the last definition added in the composition as **AA** you have referred to it as **AAB** for someone else who will be making modification, can we maintain it as **AA**

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxRttQueries.java [hunk 2]
```diff
@@ -346,18 +346,19 @@ public static String getTreatmentInterruptionOfXDaysBeforeReturningToTreatmentQu
...
     if (minDays == null && maxDays != null) {
       query +=
-          "    WHERE  TIMESTAMPDIFF(day, B.earliest_date, query_a.final_encounter_date) < ${maxDays} ";
+          "    WHERE  TIMESTAMPDIFF(day, B.final_earliest_date, query_a.final_encounter_date) < ${maxDays} ";
     } else if (minDays != null && maxDays == null) {
       query +=
-          "    WHERE  TIMESTAMPDIFF(day, B.earliest_date, query_a.final_encounter_date) >= ${minDays} ";
-    } else {
+          "    WHERE  TIMESTAMPDIFF(day, B.final_earliest_date, query_a.final_encounter_date) >= ${minDays} ";
+    } else if (minDays != null && maxDays != null) {
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxRttQueries.java]: At this point already **Condition 'maxDays != null' is always 'true' when reached** maybe just remove it

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java (hunk 1 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java:603]: Actually the specs calls **Date from AA (the most recent one, and minus 28 days) ) minus Date from B (the earliest one)** so I have merged both queries into one and referred to it as **AAB**

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxRttQueries.java (hunk 2 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxRttQueries.java]: Done removing the unnecessary condition check

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java (hunk 1 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java:603]: OK, that is clear

**Review by ningosi — APPROVED** [review]

**Pull request merged**